lib/sentry/transaction.ex

```diff
@@ -0,0 +1,75 @@
+defmodule Sentry.Transaction do
+  @type t() :: %__MODULE__{}
+
+  alias Sentry.{Config, UUID}
+
+  @enforce_keys ~w(event_id span_id spans)a
+
+  defstruct [
+    :event_id,
+    :environment,
+    :span_id,
+    :transaction,
+    :transaction_info,
+    :contexts,
+    :measurements,
+    :spans,
+    type: "transaction"
+  ]
+
+  def new(attrs) do
+    struct!(
+      __MODULE__,
+      attrs
+      |> Map.put(:event_id, UUID.uuid4_hex())
+      |> Map.put(:environment, Config.environment_name())
+    )
+  end
+
+  # Used to then encode the returned map to JSON.
+  @doc false
+  def to_map(%__MODULE__{} = transaction) do
+    transaction_attrs =
+      Map.take(transaction, [
+        :event_id,
+        :environment,
+        :transaction,
+        :transaction_info,
+        :contexts,
+        :measurements,
+        :type
+      ])
+
+    {[root_span], child_spans} = Enum.split_with(transaction.spans, &is_nil(&1.parent_span_id))
```

Comment:
I’m not super comfy with this. I think we should make illegal states unrepresentable if possible, so possibly have a `:root_span` and `:child_spans` fields maybe?